pkg/controller/json_bind.go

```diff
@@ -35,6 +35,9 @@ func BindJSON(w http.ResponseWriter, r *http.Request, data interface{}) error {
 	if !IsJSONContentType(r) {
 		return fmt.Errorf("content-type is not application/json")
 	}
+	if r.Body == nil {
+		return fmt.Errorf("no body set")
```

Comment:
I think there might be cases where it's valid to have an empty json body

Reply:
moving a line around. The defer r.Body.Close() panics. `defer` is tricky